packages/cli/src/eventbus/MessageEventBusDestination/MessageEventBusDestinationSentry.ee.ts

```diff
@@ -0,0 +1,142 @@
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+/* eslint-disable @typescript-eslint/no-unsafe-return */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+import { MessageEventBusDestination } from './MessageEventBusDestination.ee';
+import * as Sentry from '@sentry/node';
+import { eventBus } from '../MessageEventBus/MessageEventBus';
+import {
+	LoggerProxy,
+	MessageEventBusDestinationOptions,
+	MessageEventBusDestinationSentryOptions,
+	MessageEventBusDestinationTypeNames,
+} from 'n8n-workflow';
+import { GenericHelpers } from '../..';
+import { isLogStreamingEnabled } from '../MessageEventBus/MessageEventBusHelper';
+import { EventMessageTypes } from '../EventMessageClasses';
+import { eventMessageGenericDestinationTestEvent } from '../EventMessageClasses/EventMessageGeneric';
+
+export const isMessageEventBusDestinationSentryOptions = (
+	candidate: unknown,
+): candidate is MessageEventBusDestinationSentryOptions => {
+	const o = candidate as MessageEventBusDestinationSentryOptions;
+	if (!o) return false;
+	return o.dsn !== undefined;
+};
+
+export class MessageEventBusDestinationSentry
+	extends MessageEventBusDestination
+	implements MessageEventBusDestinationSentryOptions
+{
+	dsn: string;
+
+	tracesSampleRate = 1.0;
+
+	sendPayload: boolean;
+
+	sentryInitSuccessful = false;
+
+	sentryClient: Sentry.NodeClient;
+
+	constructor(options: MessageEventBusDestinationSentryOptions) {
+		super(options);
+		// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+		this.label = options.label ?? 'Sentry DSN';
+		this.__type = options.__type ?? MessageEventBusDestinationTypeNames.sentry;
+		this.dsn = options.dsn;
+		if (options.sendPayload) this.sendPayload = options.sendPayload;
+		if (options.tracesSampleRate) this.tracesSampleRate = options.tracesSampleRate;
+		const { ENVIRONMENT: environment } = process.env;
+
+		GenericHelpers.getVersions()
+			.then((versions) => {
+				this.sentryClient = new Sentry.NodeClient({
+					dsn: this.dsn,
+					tracesSampleRate: this.tracesSampleRate,
+					environment,
+					release: versions.cli,
+					transport: Sentry.makeNodeTransport,
+					integrations: Sentry.defaultIntegrations,
+					stackParser: Sentry.defaultStackParser,
+				});
+				LoggerProxy.debug(`MessageEventBusDestinationSentry with id ${this.getId()} initialized`);
+				this.sentryInitSuccessful = true;
+			})
+			.catch((error) => {
+				this.sentryInitSuccessful = false;
+				console.error(error);
+			});
+	}
+
+	async receiveFromEventBus(msg: EventMessageTypes): Promise<boolean> {
+		let sendResult = false;
+		if (!this.sentryInitSuccessful) return sendResult;
+		if (msg.eventName !== eventMessageGenericDestinationTestEvent) {
+			if (!isLogStreamingEnabled()) return sendResult;
```

Comment:
This if is duplicated for every Destination - can we perhaps add this check to the caller?

Reply:
This check is in there to prevent someone from sending a message to the destination directly, circumventing a disabled logStreaming